nrf-hal-common/src/saadc.rs

```diff
@@ -74,23 +80,129 @@ impl Saadc {
         saadc.samplerate.write(|w| w.mode().task());
 
-        saadc.ch[0].config.write(|w| {
+        // Calibrate
+        saadc.tasks_calibrateoffset.write(|w| unsafe { w.bits(1) });
+        while saadc.events_calibratedone.read().bits() == 0 {}
+
+        Saadc { saadc }
+    }
+
+    pub fn channel(&mut self, n: usize) -> Channel {
+        Channel {
+            saadc: &mut self.saadc,
+            channel: n,
+        }
+    }
+}
+
+pub struct Channel<'a> {
+    saadc: &'a mut SAADC,
+    channel: usize,
+}
+
+impl<'a> Channel<'a> {
+    pub fn configure(
+        &mut self,
+        pseln: impl Into<Pseln>,
+        pselp: impl Into<Pselp>,
+        config: SaadcChannelConfig,
+    ) {
+        let SaadcChannelConfig {
+            reference,
+            gain,
+            mode,
+            resistor,
+            time,
+        } = config;
+
+        self.saadc.ch[self.channel].config.write(|w| {
             w.refsel().variant(reference);
             w.gain().variant(gain);
             w.tacq().variant(time);
-            w.mode().se();
+            w.mode().variant(mode);
             w.resp().variant(resistor);
             w.resn().bypass();
             w.burst().enabled();
             w
         });
-        saadc.ch[0].pseln.write(|w| w.pseln().nc());
 
-        // Calibrate
-        saadc.tasks_calibrateoffset.write(|w| unsafe { w.bits(1) });
-        while saadc.events_calibratedone.read().bits() == 0 {}
+        self.saadc.ch[self.channel]
+            .pselp
+            .write(|w| w.pselp().variant(pselp.into()));
+        self.saadc.ch[self.channel]
+            .pseln
+            .write(|w| w.pseln().variant(pseln.into()));
+    }
+
+    /// Sample channel `PIN` for the configured ADC acquisition time in differential input mode.
+    /// Note that this is a blocking operation.
+    pub fn read(&mut self) -> nb::Result<i16, ()> {
+        let mut val: i16 = 0;
+        self.saadc
+            .result
+            .ptr
+            .write(|w| unsafe { w.ptr().bits(((&mut val) as *mut _) as u32) });
+        self.saadc
+            .result
+            .maxcnt
+            .write(|w| unsafe { w.maxcnt().bits(1) });
+
+        // Conservative compiler fence to prevent starting the ADC before the
+        // pointer and maxcount have been set.
+        compiler_fence(SeqCst);
+
+        self.saadc.tasks_start.write(|w| unsafe { w.bits(1) });
+        self.saadc.tasks_sample.write(|w| unsafe { w.bits(1) });
+
+        while self.saadc.events_end.read().bits() == 0 {}
+        self.saadc.events_end.reset();
 
-        Saadc(saadc)
+        // Will only occur if more than one channel has been enabled.
+        if self.saadc.result.amount.read().bits() != 1 {
+            return Err(nb::Error::Other(()));
+        }
+
+        // Second fence to prevent optimizations creating issues with the EasyDMA-modified `val`.
+        compiler_fence(SeqCst);
+
+        Ok(val)
+    }
+
+    pub fn as_micros(&self, value: i16) -> i32 {
```

Comment:
Since we use raw integers everywhere (and not something like `uom`), let's clarify that this returns microvolts and has nothing to do with `Duration::as_micros`

```suggestion
    /// Converts a raw reading from this channel to µV.
    pub fn to_microvolts(&self, value: i16) -> i32 {
```